File: app/src/main/java/com/kirchhoff/example/githubclient/repository/keyvalue/HawkKeyValueStorage.java
package com.kirchhoff.example.githubclient.repository.keyvalue;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.orhanobut.hawk.Hawk;

public class HawkKeyValueStorage implements KeyValueStorage {

    @Nullable
    private static HawkKeyValueStorage INSTANCE = null;

    private HawkKeyValueStorage() {
    }

    public static HawkKeyValueStorage getInstance() {
        if (INSTANCE == null) {
            INSTANCE = new HawkKeyValueStorage();
        }

        return INSTANCE;
    }

    @Override
    public void saveToken(@NonNull String token) {
        Hawk.put(TOKEN_KEY, token);
    }

    @NonNull
    @Override
    public String getToken() {
        return Hawk.get(TOKEN_KEY, "");
    }

    @Override
    public void saveUserName(@NonNull String userName) {
        Hawk.put(USER_NAME_KEY, userName);
    }

    @NonNull
    @Override
    public String getUserName() {
        return Hawk.get(USER_NAME_KEY, "");
    }

    @Override
    public void clear() {
        Hawk.clear();
    }
}

File: app/src/main/java/com/kirchhoff/example/githubclient/repository/keyvalue/HawkKeyValueStorage.kt
package com.kirchhoff.example.githubclient.repository.keyvalue


import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage.Companion.TOKEN_KEY
import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage.Companion.USER_NAME_KEY
import com.orhanobut.hawk.Hawk

/**
 * @author Kirchhoff-
 */

object HawkKeyValueStorage : KeyValueStorage {

    override fun saveToken(token: String) {
        Hawk.put(TOKEN_KEY, token)
    }

    override val token: String
        get() = Hawk.get(TOKEN_KEY, "")

    override fun saveUserName(userName: String) {
        Hawk.put(USER_NAME_KEY, userName)
    }

    override val userName: String
        get() = Hawk.get(USER_NAME_KEY, "")

    override fun clear() {
        Hawk.clear()
    }

}
